src/Transport/Sending/DefaultBatchedOperationsToBrokeredMessagesConverter.cs

```diff
@@ -66,7 +66,27 @@ void SetReplyToAddress(OutgoingMessage outgoingMessage, BrokeredMessage brokered
         {
             if (outgoingMessage.Headers.ContainsKey(Headers.ReplyToAddress))
             {
-                brokeredMessage.ReplyTo = mapper.Map(outgoingMessage.Headers[Headers.ReplyToAddress]);
+                var replyTo = mapper.Map(outgoingMessage.Headers[Headers.ReplyToAddress]);
+
+                // ensure that the reply to address contains namespace info in case the message is sent to destination only namespace.
+                if (!replyTo.HasSuffix)
+                {
+                    var namespaces = settings.Get<NamespaceConfigurations>(WellKnownConfigurationKeys.Topology.Addressing.Partitioning.Namespaces);
+                    var defaultName = settings.Get<Func<string>>(WellKnownConfigurationKeys.Topology.Addressing.DefaultNamespaceName)();
+                    var selected = namespaces.FirstOrDefault(ns => ns.Name == defaultName);
+                    if (selected == null)
+                    {
+                        selected = namespaces.FirstOrDefault(ns => ns.Purpose == NamespacePurpose.Partitioning);
+                    }
+
+                    if (selected != null)
+                    {
+                        replyTo = mapper.Map(new EntityAddress(replyTo.Name, selected.Name));
+                    }
+                }
+
+                outgoingMessage.Headers[Headers.ReplyToAddress] = replyTo;
```

Comment:
yes it is, but the headers are also copied over into brokered message properties later